src/Main/Settings.cpp

```diff
@@ -1,144 +1,73 @@
 
-using namespace FOEDAG;
+This program is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
 
-Settings::Settings() {}
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
 
-void Settings::setValue(QJsonObject& parent, const QString& key,
-                        const QJsonValue& value) {
-  parent.insert(key, value);
-}
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
 
-void Settings::setValue(const QString& key, const QJsonValue& value) {
-  setValue(m_data, key, value);
-}
-
-void Settings::setString(const QString& key, const QString& string) {
-  QJsonValue value = QJsonValue(string);
-  setValue(key, value);
-}
-
-void Settings::updateJson(const QString& key, QJsonObject& newJson) {
-  updateJson(m_data, key, newJson);
-}
-
-void Settings::updateJson(QJsonObject& parent, const QString& key,
-                          QJsonObject& newJson) {
-  QJsonValue val = parent.value(key);
+#include "Settings.h"
 
-  // Change this value to an object if it isn't one already
-  // Whatever the former value was will be lost
-  if (!val.isObject()) {
-    setValue(parent, key, QJsonObject{});
-  }
+#include <QFile>
+#include <iostream>
 
-  // Qt currently returns a value instead of a reference when calling toObject,
-  // as such we can't updated nested JSON objects and will have to manually set
-  // the new values after they are updated.
-  // https://stackoverflow.com/a/29361151
-  QJsonObject parentObj =
-      parent.value(key).toObject();  // toObject call should be superfluous from
-                                     // previous line
-
-  // Step through each new key
-  for (const QString& newKey : newJson.keys()) {
-    QJsonValue val = parentObj.value(newKey);
-    if (!val.isObject()) {
-      QJsonValue newVal = newJson.value(newKey);
-      if (newVal.isObject()) {
-        QJsonObject obj = newVal.toObject();
-        updateJson(parentObj, newKey, obj);
-      } else {
-        setValue(parentObj, newKey, newVal);
-      }
-    } else {
-      // add QJsonObject recursion
-      QJsonObject tempObj = newJson.value(newKey).toObject();
-      updateJson(parentObj, newKey, tempObj);
-      ;
-    }
-  }
+using namespace FOEDAG;
 
-  // Manually update json
-  parent.insert(key, parentObj);
-}
+Settings::Settings() { loadSettings(); }
 
-QJsonValue Settings::get(QJsonObject& parent, const QString& key) {
-  return parent.value(key);
+void Settings::loadSettings() {
+  QStringList jsonFiles{
+      "/usr/local/share/foedag/etc/settings/settings_test.json",
```

Comment:
it is better to hardcore file names outside the class. e.g. when you create object.

Reply:
Good call, I've moved the loadSettings into Foedag.cpp::Foedag() (figured putting it in main.cpp::main() was a bit too low level).